### DISCUSSION TITLE
Julia concepts we want to learn

### DISCUSSION THREAD

**Discussion by suvayu** [TulipaEnergy/TulipaEnergyModel.jl#320] [Ideas]
Can we collect all the Julia concepts we want to learn in one single thread?  Then it might be easier for @abelsiqueira to cover them in the 1 hr learning discussion that we were pondering last week.  We can populate this list as we encounter them

My wish list:
- error handling & propagation
- reflection (inspecting the current Julia session)
  - it would be nice to have this in the toolbox because JuMP introduces many new symbols in the current session. This will help us discover them and learn faster; e.g. by accident I learnt there are names available to check termination status: `OPTIMAL` & `INFEASIBLE`.  Knowing this helps in writing code with good error handling.
- overview of any functional programming API Julia might have, stuff like `map`, `filter`, `first`, etc - great for working with arrays and matrices

**Comment by clizbe** [comment]
Ooh nice! If we still want this, I would be interested in data structures, their differences, how data accessed (although I think @abelsiqueira has done a good job of documenting the stuff we're using).

**Comment by clizbe** [comment]
Is this still something we want? If so maybe we can spend a half-day on it soon.

**Reply by datejada** [reply]
Let's close this discussion for now and double-check when new members join the team what extra topics are of interest.
